src/main/java/de/hpi/bpt/argos/api/entity/EntityEndpointImpl.java

```diff
@@ -0,0 +1,307 @@
+    private JsonObject getEventTypeJson(EventType eventType) {
+        try {
+            JsonObject jsonEventType = new JsonObject();
+
+            jsonEventType.addProperty("Id", eventType.getId());
+            jsonEventType.addProperty("Name", eventType.getName());
+            jsonEventType.addProperty("NumberOfEvents",
+                    PersistenceAdapterImpl.getInstance().getEventCountOfEventType(eventType.getId()));
+            jsonEventType.addProperty("TimestampAttributeId", eventType.getTimeStampAttributeId());
+
+            return jsonEventType;
+        } catch (Exception e) {
+            LoggerUtilImpl.getInstance().error(logger, "cannot parse event type", e);
+            return new JsonObject();
+        }
+    }
+
+    /**
+     * This method returns event types as a JsonObject.
+     * @param eventTypes - the event types
+     * @return - a json representation of the event types
+     */
+    private JsonObject getEventTypesJson(List<EventType> eventTypes) {
+        JsonArray eventTypesJson = new JsonArray();
+        for (EventType eventType : eventTypes) {
+            eventTypesJson.add(getEventTypeJson(eventType));
+        }
+        return eventTypesJson.getAsJsonObject();
+    }
+
+    /**
+     * This method returns events as a JsonObject.
+     * @param events - the events
+     * @return - a json representation of the events
+     */
+    private JsonObject getEventsJson(List<Event> events) {
+        JsonArray eventsJson = new JsonArray();
+        for (Event event : events) {
+            List<Attribute> attributes = PersistenceAdapterImpl.getInstance().getAttributes(event.getId());
+            JsonArray jsonAttributes = getAttributesJson(attributes);
+            eventsJson.add(jsonAttributes);
+        }
+        return eventsJson.getAsJsonObject();
+    }
+
+    /**
+     * This method returns attributes as a JsonObject.
+     * @param attributes - the attributes
+     * @return - a json representation of the attributes
+     */
+    private JsonArray getAttributesJson(List<Attribute> attributes) {
+        JsonArray jsonAttributes = new JsonArray();
+        for (Attribute attribute : attributes) {
+            JsonObject jsonAttribute = new JsonObject();
+            TypeAttribute typeAttribute = PersistenceAdapterImpl.getInstance().getTypeAttribute(attribute.getTypeAttributeId());
+            jsonAttribute.addProperty("Name", typeAttribute.getName());
+            jsonAttribute.addProperty("Value", attribute.getValue());
+            jsonAttributes.add(jsonAttribute);
+        }
+        return jsonAttributes;
+    }
+
+    /**
+     * This method returns the id given in request.
+     * @param request the request with entity id
+     * @return id of the entity in request
+     */
+    private long getEntityId(Request request) {
+        return endpointUtil.validateLong(
+                request.params(EntityEndpoint.getEntityIdParameter(false)),
+                (Long input) -> input > 0);

```

Comment:
Since our virtualRoot Node will have a negative Id, should be `(Long input) -> input != 0` for example.